tests/func/test_listing.py

```diff
@@ -0,0 +1,34 @@
+from datachain.client.local import FileClient
+from datachain.lib.dc import DataChain
+from datachain.lib.listing import list_bucket
+from tests.data import ENTRIES
+
+
+def test_listing_generator(cloud_test_catalog, cloud_type):
+    ctc = cloud_test_catalog
+
+    uri = f"{ctc.src_uri}/cats"
+
+    dc = DataChain.from_records(DataChain.DEFAULT_FILE_RECORD).gen(
+        file=list_bucket(uri, client_config=ctc.catalog.client_config)
+    )
+    assert dc.count() == 2
+
+    entries = sorted(
+        [e for e in ENTRIES if e.path.startswith("cats/")], key=lambda e: e.path
+    )
+    files = sorted(dc.collect("file"), key=lambda f: f.path)
+
+    for cat_file, cat_entry in zip(files, entries):
+        if cloud_type == "file":
```

Comment:
It looks like we can avoid this. Otherwise, we will be always treating local files as special type which will lead to tons of extra logic.

Ideally, the records should be compared like `assert cat_file.model_dump() == cat_entry.model_dump()` or just `assert files == entries` for the whole list.

Is it possible to achive this?

Reply:
Yea ... that is one downside of current file system listing implementation where source is always `file:///` ... upside is that partial indexing works and no need for that `add_storage` method. I will think how to re-implement it again now with new listing in API since we should get rid of partial indexing anyway. I think the biggest question is what is exactly `source` field for listing rows .... with s3 bucket it's clear - it's the name of the bucket itself, but with local FS it's not exactly clear.